src/ddmd/statementsem.d

```diff
@@ -483,6 +490,366 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
         result = fs;
     }
 
+    private static template MakeTupleForeachRet(bool isDecl){
+        static if(isDecl)
+        {
+            alias MakeTupleForeachRet = Dsymbols*;
+        }
+        else
+        {
+            alias MakeTupleForeachRet = void;
+        }
+    }
+
+    MakeTupleForeachRet!isDecl makeTupleForeach(bool isStatic, bool isDecl)(ForeachStatement fs, TupleForeachArgs!(isStatic, isDecl) args)
```

Comment:
The function contains mostly existing code that was not written by me. I can comment the parts I have added to handle `static foreach`. The point of the function is to share the code of `foreach` over a tuple and `static foreach`.